ecs-agent/acs/session/attach_resource_responder.go

```diff
@@ -180,9 +181,23 @@ func validateAttachmentAndReturnProperties(message *ecsacs.ConfirmAttachmentMess
 		attachmentProperties[name] = value
 	}
 
-	err = resource.ValidateResource(attachmentProperties)
+	// For "AmazonElasticBlockStorage" used by the EBS attach, ACS is using attachmentType to indicate its attachment type.
+	attachmentType := aws.StringValue(message.Attachment.AttachmentType)
+	if attachmentType == resource.AmazonElasticBlockStorage {
+		err = resource.ValidateRequiredProperties(
+			attachmentProperties,
+			resource.GetVolumeSpecificPropertiesForEBSAttach(),
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "resource attachment validation by attachment type failed")
+		}
+		return attachmentProperties, nil
+	}
+
+	// For "EphemeralStorage" and "ElasticBlockStorage", ACS is using resourceType to indicate its attachment type.
```

Comment:
we have different ways to differentiate EBS now, correct?
`AttachmentType ` vs `resourceType`.  Can we just call it "Legacy EBS" wherever we're pulling the type from the resource?

Reply:
Yes.
And I can update these comments in the next PR.